core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -331,6 +333,9 @@ public GetResult get(Get get, Function<String, Searcher> searcherFactory) throws
         try (ReleasableLock lock = readLock.acquire()) {
             ensureOpen();
             if (get.realtime()) {
+                if (isRealtimeGetEnabled == false) {
+                    throw new IllegalArgumentException("get with realtime=true is disabled on this index");
```

Comment:
Since the person who configures the cluster might not be the same as the one who sends GET requests, I think I like better that we make things explicit and require users to pass realtime=false.